sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/NonAzureOpenAIClientImpl.java

```diff
@@ -0,0 +1,721 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai.implementation;
+
+import com.azure.ai.openai.models.ChatCompletionsOptions;
+import com.azure.ai.openai.models.CompletionsOptions;
+import com.azure.ai.openai.models.EmbeddingsOptions;
+import com.azure.core.annotation.BodyParam;
+import com.azure.core.annotation.ExpectedResponses;
+import com.azure.core.annotation.HeaderParam;
+import com.azure.core.annotation.Host;
+import com.azure.core.annotation.HostParam;
+import com.azure.core.annotation.Post;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceInterface;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.annotation.UnexpectedResponseExceptionType;
+import com.azure.core.exception.ClientAuthenticationException;
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.exception.ResourceModifiedException;
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.RestProxy;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.serializer.SerializerAdapter;
+import reactor.core.publisher.Mono;
+
+/**
+ * Implementation for calling Non-Azure OpenAI service
+ */
+public final class NonAzureOpenAIClientImpl {
```

Comment:
Is there a difference in the client impl for non-Azure and Azure services? If the only change is hardcoded endpoint, we should look into merging the two with a configurable endpoint.

Reply:
Yeah. Which should be redesigned with changes in TypeSpec. For the beta. I try to no make changes to autogen imple.